conan.lock

```diff
@@ -0,0 +1,64 @@
+        "libmount/2.39.2#64850679fa9d5ead878e5076b0cab2a8%1705591482.625",
+        "libcap/2.69#7ef2d60864c2e58e89db957be936dc49%1693427138.461",
+        "libbacktrace/cci.20210118#a7691bfccd8caaf66309df196790a5a1%1722218217.276",
+        "grpc/1.69.0#937b22d3f20a2fdd2f2d278afadb8999%1742893931.477",
+        "c-ares/1.34.3#55aa867c731f23300cd544f6d90a794d%1731428505.375",
+        "bzip2/1.0.8#d00dac990f08d991998d624be81a9526%1724661266.998",
+        "boost/1.86.0#ce76e7477e466d7d8cbcf738c5d64175%1736197454.755",
+        "abseil/20240116.2#dfc978c35194fee939d236253044b967%1740151184.642"
+    ],
+    "build_requires": [
+        "zstd/1.5.5#1f239731dc45147c7fc2f54bfbde73df%1715599909.17",
+        "zlib/1.3.1#b8bc2603263cf7eccbd6e17e66b0ed76%1733936244.862",
+        "xz_utils/5.4.5#b885d1d79c9d30cff3803f7f551dbe66%1724318972.064",
+        "re2/20230301#e8d673155a0494f27c7589b611e8350d%1709641579.83",
+        "protobuf/5.27.0#1fc80ae5667e6cd1d740ca84679b17d6%1742892879.34",
+        "pkgconf/2.2.0#6462942a22803086372db44689ba825f%1713364853.749",
+        "pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605%1701537936.436",
+        "pcre2/10.42#ac2227071f79a062167cf2fee7189d66%1727865720.942",
+        "openssl/3.4.1#85da54eb530cab53de0cc0ee9065d066%1741868597.788",
+        "ninja/1.12.1#fd583651bf0c6a901943495d49878803%1724079877.609",
+        "meson/1.4.0#2262941cc8fbb0099dd0c196ca2a6c01%1726730116.631",
+        "meson/1.2.2#21b73818ba96d9eea465b310b5bbc993%1726730120.212",
+        "m4/1.4.19#b38ced39a01e31fef5435bc634461fd2%1700758725.451",
+        "lz4/1.9.4#652b313a0444c8b1d60d1bf9e95fb0a1%1721807480.477",
+        "libxcrypt/4.4.36#4b4e8f20794f1997dd59eeed0b7cdcfb%1696519152.422",
+        "libtool/2.4.7#a182d7ce8d4c346a19dbd4a5d532ef68%1742900203.747",
+        "libsystemd/255.10#0552f8de4d3eeaaefd87f6677dfbce1f%1723640836.911",
+        "libselinux/3.6#5a78ff6ae5034eeaac8da723361a8ce4%1717655459.344",
+        "libmount/2.39.2#64850679fa9d5ead878e5076b0cab2a8%1705591482.625",
+        "libcap/2.69#7ef2d60864c2e58e89db957be936dc49%1693427138.461",
+        "grpc/1.69.0#937b22d3f20a2fdd2f2d278afadb8999%1742893931.477",
+        "gperf/3.1#1d622ad9717e9348ed3685c9994ad0b9%1709324989.76",
+        "gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b%1701248168.479",
+        "flex/2.6.4#e35bc44b3fcbcd661e0af0dc5b5b1ad4%1674818991.113",
+        "cmake/3.31.6#ed0e6c1d49bd564ce6fed1a19653b86d%1740594968.467",
+        "c-ares/1.34.3#55aa867c731f23300cd544f6d90a794d%1731428505.375",
+        "bzip2/1.0.8#d00dac990f08d991998d624be81a9526%1724661266.998",
+        "b2/5.2.1#91bc73931a0acb655947a81569ed8b80%1718416612.241",
+        "automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50%1688481772.751",
+        "autoconf/2.71#f9307992909d7fb3df459340f1932809%1711983104.648",
+        "abseil/20250127.0#7f3643d6f847196fe0bdc2b83c109831%1742212255.256",
+        "abseil/20240116.2#dfc978c35194fee939d236253044b967%1740151184.642"
```

Comment:
This is very interesting, and is *maybe* concerning, though I can't quite tell what the implications for having two versions of `abseil` in `build_requires` are w.r.t. to the scary ODR consequences of having two abseils in the graph. Hopefully @lia-viam can provide further insights.

I note though that there are other duplications in this list, particularly `meson` and `pkgconf`, which do make more sense as things that you could have two of without issue since they are presumably only used at build time.

Reply:
Stuff in the `build_requires` section is only used in tests and compile time tools (e.g., the `cmake` and `pkgconf` packages). 

I've tried running `conan lock create . -vtrace`, but haven't yet been able to track down where these dependencies are coming from. but my guess is that we have two tools used at build time that depend on the two different versions.

The ones I'd be concerned about are up in the `requires` section, and they're all unique.

Reply:
Okay, `conan graph info` is more useful! `protobuf` depends on any version of `abseil` between 20230802.1 and 20250127.0, whereas literally everything else (including `grpc`, oddly) depends on the much narrower range 20240116.1 to 20240117.0. The issue is just that `protobuf` is being very permissive with which versions it accepts.

My hope is that `conan` is smart enough to realize that a single version can satisfy all these requirements, though I don't actually know.